File: src/main/java/ar/edu/um/haberes/rest/model/Categoria.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Categoria extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -7627833487679606596L;

	@Id
	private Integer categoriaId;

	private String nombre = "";
	private BigDecimal basico = BigDecimal.ZERO;
	private Byte docente = 0;
	private Byte noDocente = 0;
	private Byte liquidaPorHora = 0;

}

File: src/main/java/um/haberes/rest/kotlin/model/Categoria.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import um.haberes.rest.model.Auditable
import java.math.BigDecimal

@Entity
@Table
data class Categoria(

    @Id
    var categoriaId: Int? = null,

    var nombre: String = "",
    var basico: BigDecimal = BigDecimal.ZERO,
    var docente: Byte = 0,
    var noDocente: Byte = 0,
    var liquidaPorHora: Byte = 0

) : Auditable()
